lib/events.js

```diff
@@ -57,7 +57,7 @@ api.emit = function(event /* ... */) {
 
   // default callback
   let callback = () => {};
-  if(typeof args[args.length - 1] === 'function') {
+  if(args.length === 3 && typeof args[args.length - 1] === 'function') {
```

Comment:
Perhaps this will do it instead? We could assume that only a regular function may be used as a callback, not a complex function object, which instead we will assume is data meant for the event.

```suggestion
  if(typeof args[args.length - 1] === 'function' && Object.keys(args[args.length - 1]).length > 0) {
```